osu.Game/Screens/Play/HUDOverlay.cs

```diff
@@ -158,7 +158,7 @@ public HUDOverlay([CanBeNull] DrawableRuleset drawableRuleset, IReadOnlyList<Mod
                     Direction = FillDirection.Vertical,
                     Children = new Drawable[]
                     {
-                        HoldToQuit = CreateHoldForMenuButton(),
+                        HoldToQuit = CreateHoldForMenuButton(touchDevice: mods.Any(m => m is ModTouchDevice)),
```

Comment:
Rather than doing this, use `SessionStatics` from within the class, don't pass anything in.

https://github.com/ppy/osu/blob/63a0ea54109c93984501b66435febf676663cab4/osu.Game/Screens/Play/PlayerTouchInputDetector.cs#L24-L29

Reply:
Please don't self resolve issues. This isn't even fixed.

Reply:
I misunderstood it as "dont pass anything to `CreateHoldForMenuButton`", I assume you meant inject it from within `HoldForMenuButton`?